CONTRIBUTING.md

```diff
@@ -301,6 +333,20 @@ sizeable amount of time, so care must be taken not to overdo it:
 1. All algorithm tests must be on very simple datasets and as quick as possible
 2. We try not to trigger CI pipelines when unnecessary (see [Skipping CI
 runs](#skipping-ci-runs)).
+3. We split the tests based on their duration into groups and run them in parallel.
+  
+   For that we use [pytest-split](https://jerry-git.github.io/pytest-split)
+   to first store the duration of all tests with `pytest --store-durations pytest --slow-tests`
+   in a `.test_durations` file.
+   
+   Then we can have as many splits as we want:
```

Comment:
How are the commands below to be interpreted? I suppose each one is to be run in a different machine, because the splits into groups are deterministic, correct?

Reply:
I updated CONTRIBUTING.md with more detail.

Yes, each command is supposed to run in a separate job and yes the splits should be deterministic.